### PR TITLE
Feature security

### PR DESCRIPTION
**Pull request by Raymundogs**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/security/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    @Override
+    @Bean
+    public AuthenticationManager authenticationManagerBean() throws Exception {
+        return super.authenticationManagerBean();
+    }
+
+    /**
+     * Encodes the password
+     * return password encoded with BCrypt
```
Comment (JessickaMoya) [inline src/main/java/org/fundacion/jala/converter/security/SecurityConfig.java]: I think it would be @return instead of only return

Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/security/SecurityConfig.java]: I corrected it.

**Review by JessickaMoya — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/controller/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * Create an authentication response with the token
+     * @param authenticationRequest a request to authenticate
+     * @return response entity with the token
+     * @throws Exception when not found
+     */
+    @RequestMapping(value = "/authenticate", method = RequestMethod.POST)
+    public ResponseEntity<?> createAuthenticationToken(@RequestBody final AuthenticationRequest authenticationRequest) throws Exception {
+        try {
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/AuthController.java:48]: The loggers are missing

**Review by Paola-Xaq — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/models/AuthenticationRequest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
+package org.fundacion.jala.converter.models;
+
+public class AuthenticationRequest {
+    private String username;
+    private String password;
+
+    public AuthenticationRequest() {
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/models/AuthenticationRequest.java]: Maybe you should delete this constructor

File: src/main/java/org/fundacion/jala/converter/models/AuthenticationResponse.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
+package org.fundacion.jala.converter.models;
+
+public class AuthenticationResponse {
+    private final String jwt;
+
+    public AuthenticationResponse(String jwt) {
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/models/AuthenticationResponse.java]: Your parameters in some methods should be final and shoul not have the same name as the variable to which it is being assigned

File: src/main/java/org/fundacion/jala/converter/controller/AuthController.java (hunk 2 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/AuthController.java:48]: Added the loggers.

File: src/main/java/org/fundacion/jala/converter/models/AuthenticationRequest.java (hunk 3 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/models/AuthenticationRequest.java]: I deleted the constructor.

File: src/main/java/org/fundacion/jala/converter/models/AuthenticationResponse.java (hunk 4 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/models/AuthenticationResponse.java]: I do not agree on the variable's names, but I changed them.

**Review by danielaSantaCruz — APPROVED** [review]

**Pull request merged**